core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1089,7 +1089,7 @@ class DAGScheduler(
         // To avoid UI cruft, ignore cases where value wasn't updated
         if (acc.name.isDefined && !updates.isZero) {
           stage.latestInfo.accumulables(id) = acc.toInfo(None, Some(acc.value))
-          event.taskInfo.accumulables += acc.toInfo(Some(updates.value), Some(acc.value))
+          event.taskInfo.accumulables ::= acc.toInfo(Some(updates.value), Some(acc.value))
```

Comment:
Something I was a bit unsure of but want to revisit now:

It looks like the loop 

```
 for (accumulableInfo <- info.accumulables) {
        stageData.accumulables(accumulableInfo.id) = accumulableInfo
      }
```

in `JobProgressListener.onTaskEnd()` may assume that accumulator updates are appended to the list so that the newest update for a particular accumulator appears last, but _prepending_ here will change that. I don't think that this behavior is covered by any existing unit tests, though. Instead of using a list here, I might want to use an immutable `Queue`. Let me go ahead and make this change now.